spec/components/list_spec.rb

```diff
@@ -0,0 +1,89 @@
+require "rails_helper"
+
+describe "List", type: :view do
+  def component_name
+    "list"
+  end
+
+  it "does not render anything if no data is passed" do
+    test_data = {}
+
+    assert_empty render_component(test_data)
+  end
+
+  it "does not render anything if nothing is in the items array" do
+    test_data = { items: [] }
+
+    assert_empty render_component(test_data)
+  end
+
+  it "renders an unordered list by default" do
+    render_component(
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ul.govuk-list li", text: "Test item"
+    assert_select "ul.govuk-list li:nth-child(2)", text: "Another test item"
+  end
+
+  it "adds an aria-label" do
+    render_component(
+      aria_label: "An aria-label to give this context.",
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ul[aria-label]"
+    assert_select "ul[aria-label='An aria-label to give this context.']"
+  end
+
+  it "renders an ordered list" do
+    render_component(
+      list_type: "number",
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ol.govuk-list li", text: "Test item"
+    assert_select "ol.govuk-list li:nth-child(2)", text: "Another test item"
+  end
+
+  it "renders an unordered list with visible bullets" do
+    render_component(
+      visible_counters: true,
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ul.govuk-list--bullet li", text: "Test item"
+    assert_select "ul.govuk-list--bullet li:nth-child(2)", text: "Another test item"
+  end
+
+  it "renders an ordered list with visible counters" do
+    render_component(
+      list_type: "number",
+      visible_counters: true,
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ol.govuk-list--number li", text: "Test item"
+    assert_select "ol.govuk-list--number li:nth-child(2)", text: "Another test item"
+  end
+
+  it "adds extra spacing to each list item" do
+    render_component(
+      extra_spacing: true,
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ul.govuk-list--spaced li", text: "Test item"
+    assert_select "ul.govuk-list--spaced li:nth-child(2)", text: "Another test item"
+  end
+
+  it "adds an `id` correctly" do
+    render_component(
+      id: "this-is-a-test-id",
+      items: ["Test item", "Another test item"],
+    )
+
+    assert_select "ul#this-is-a-test-id li", text: "Test item"
+    assert_select "ul#this-is-a-test-id li:nth-child(2)", text: "Another test item"
+  end
+end
```

Comment:
Hey @injms - a common use case is the use of links within lists. It might be worth adding a test for this (and possibly some documentation too)

Reply:
Good call - I've added an explanation at the top the documentation and a test to ensure that links are being rendered correctly. Each example has a link in it since they all use the same data - shoutout to YAMLs ability to reference itself nicely.